tests/robustness/validate/watch.go

```diff
@@ -138,8 +139,46 @@ func validateResumable(t *testing.T, events []model.WatchEvent, report report.Cl
 		}
 		firstEvent := firstWatchEvent(op)
 		// If watch is resumable, first event it gets should the first event that happened after the requested revision.
-		if firstEvent != nil && events[index] != *firstEvent {
-			t.Errorf("Resumable - A broken watch can be resumed by establishing a new watch starting after the last revision received in a watch event before the break, so long as the revision is in the history window, watch request: %+v, event missing: %+v", op.Request, events[index])
+		if firstEvent != nil && events[index].PersistedEvent != firstEvent.PersistedEvent {
+			t.Errorf("Resumable - A broken watch can be resumed by establishing a new watch starting after the last revision received in a watch event before the break, so long as the revision is in the history window, watch request: %+v, event missing: %+v, got: %+v", op.Request, events[index], *firstEvent)
+		}
+	}
+}
+
+// validatePrevKV ensures that a watch response (if configured with WithPrevKV()) returns
+// the appropriate response.
+func validatePrevKV(t *testing.T, report report.ClientReport, history []model.WatchEvent) {
+	replay := model.NewReplay(history)
+	for _, op := range report.Watch {
+		if !op.Request.WithPrevKV {
+			continue
+		}
+		for _, resp := range op.Responses {
+			for _, event := range resp.Events {
+				// get state state just before the current event.
+				state, err := replay.StateForRevision(event.Revision - 1)
+				if err != nil {
+					t.Error(err)
+				}
+
+				// This explicitly checks the condition that is used by Kubernetes
+				// while parsing events received from etcd:
+				// https://github.com/kubernetes/kubernetes/blob/a9e4f5b7862e84c4152eabe2e960f3f6fb9a4867/staging/src/k8s.io/apiserver/pkg/storage/etcd3/event.go#L59
+				if event.PrevValue == nil && !event.IsCreate {
+					t.Errorf("PrevKV - got a create event with nil prevKV %+v", event)
+				}
+				var prevValue model.ValueRevision
+				if event.PrevValue != nil {
+					prevValue = *event.PrevValue
+				}
+				// A create event will not have an entry in our history.
+				if _, ok := state.KeyValues[event.Key]; !event.IsCreate && !ok {
```

Comment:
```suggestion
				if _, prevKeyExists := state.KeyValues[event.Key]; event.IsCreate == prevKeyExists {
```

Both should cause error:
*  `IsCreate=true` and `ok=true` 
* `IsCreate=false` and `ok=false`